C/eval.c

```diff
@@ -833,26 +833,30 @@ static bool computeEvalTCOBound(memBound *dag_bound, const dag_node* dag, const
  *               outputSize == bitSize(B);
  *               output == NULL or UWORD output[ROUND_UWORD(outputSize)];
  *               input == NULL or UWORD input[ROUND_UWORD(inputSize)];
+ *               budget <= BUDGET_MAX
  *               if 'dag[len]' represents a Simplicity expression with primitives then 'NULL != env';
  */
 bool evalTCOExpression( bool *evalSuccess, flags_type anti_dos_checks, UWORD* output, ubounded outputSize, const UWORD* input, ubounded inputSize
-                      , const dag_node* dag, type* type_dag, size_t len, const txEnv* env
+                      , const dag_node* dag, type* type_dag, size_t len, ubounded budget, const txEnv* env
                       ) {
   assert(1 <= len);
   assert(len <= DAG_LEN_MAX);
+  if (BUDGET_MAX < budget) budget = BUDGET_MAX; /* Soft enforcement of budget precondition. */
```

Comment:
In a4a3ed3b0eee75521640a2c7fefc0e09127a77b5:

I think this should just be an assertion.

Reply:
done.